### PR TITLE
Throttler w/ result

### PR DESCRIPTION
**Pull request by tonikocjan**
Extended `Throttler` to support executing blocks which return result.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -73,3 +73,7 @@ fastlane/report.xml
 fastlane/Preview.html
 fastlane/screenshots/**/*.png
 fastlane/test_output
+
+#
+
+.DS_Store
```
Comment (borut-t) [inline .gitignore:79]: Do we need this?

**Review by borut-t — APPROVED** [review]
Nice addition 👌

**Review by markom-povio — APPROVED** [review]
Nice 👍

**Review by yllfejziu — APPROVED** [review]
👍

**Comment by borut-t** [comment]
@tonikocjan can we merge it?

**Pull request merged**